pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/ModelParser.groovy

```diff
@@ -188,12 +185,7 @@ class ModelParser implements Parser {
     }
 
     private @CheckForNull isFailFast(ModelASTOptions options){
-        if( options == null || options.options.isEmpty()) return false
-        def model = new ModelASTOption( new ModelASTMethodCall() )
-        model.name = "parallelsAlwaysFailFast"
-        model.args = []
-
-        return options.options.contains( model )
+        return options.options.any { it.name == "parallelsAlwaysFailFast" }
```

Comment:
You still need some null safety here - maybe `return options?.options.any ...` - sorry, I didn't think to mention this earlier.